pages/style-guide/html.md

```diff
@@ -1,77 +1,174 @@
+In general, the jQuery style guide encourages liberal spacing for improved human readability.
 
-```
+- Indentation with tabs.
+- Don't indent the direct children of `html`, `body`, `script`, or `style`. Indent everything else.
+- No whitespace at the end of line or on blank lines.
+- Separate attributes with a space.
+
+```html
+<!doctype html>
 <html lang="en">
-```
+<head>
+	<meta charset="utf-8">
+	<title>Sample Page</title>
 
-## Encoding
+	<link rel="stylesheet" href="/style.css">
+	<style>
+	body {
+		font-size: 100em;
+	}
+	</style>
+
+	<script src="/jquery.js"></script>
+	<script>
+	$( function() {
+		$( "p" ).text( $.fn.jquery );
+	} );
+	</script>
+</head>
+<body>
 
-Always define the character encoding. The encoding should be defined as early as possible.
+<section>
+	<p>jQuery is awesome!<p>
+</section>
 
+</body>
+</html>
 ```
-<meta charset="utf-8">
-```
-
-## Elements and Attributes
 
-All element and attribute names should be lowercase. Attribute values should be quoted. Optional closing tags should be included. Self-closing elements should not be closed. Optional attributes should be omitted. Always include `html`, `head`, and `body` tags.
+## Formatting
+
+ - Always use the minimal, versionless and lowercase doctype.
+ - Always define which language the page is written in.
+ - Always include `html`, `head`, and `body` tags.
+ - Always define the character encoding. Most pages should use utf-8. The encoding should be defined as early as possible and must be in the [first 1024 bytes of the document](https://html.spec.whatwg.org/multipage/semantics.html#charset).
+ - Always use lowercase names for elements and attributes.
+ - Always quote attribute values. Always use double quotes instead of single quotes.
+ - Include the optional closing tags.
+ - Self-closing elements must not be closed.
+ - Optional attributes must be omitted.
+ - `rel`, `type`, `src`, `href` and `class` attributes must be written before any other attribute.
+ - Always include an `alt` attribute for images.
+
+```html
+<!doctype html>
+<html lang="en">
+<head>
+	<meta charset="utf-8">
+	<title>Good Example</title>
+	<link rel="stylesheet" href="style.css">
+	<script src="/jquery.js"></script>
+</head>
+<body>
 
-*   No `type` or `language` attributes on `script` tags.
-*   No `type` attribute on `link` or `style` tags.
+<img class="sample-image" src="assets/img/img.png" alt="Sample Image">
+<p>This is a sample image</p>
 
+</body>
+</html>
 ```
-<script src="..."></script>
-<script></script>
-<link rel="stylesheet" href="...">
-<style></style>
-```
 
-## Indentation
+## HTML Semantics:
+
+Always use HTML elements for what they are meant to be used for.
 
-Don't indent inside `html`, `body`, `script`, or `style`. Indent inside `head` and all other elements.
+## Reducing Markup
 
-### Example
+Always try to avoid superfluous parent elements.
 
+```html
+<!-- Bad HTML -->
+<span class="avatar">
+	<img src="assets/img/img.png" alt="Sample Image">
```

Comment:
Well, this is clearly an avatar, but the code snippet shows nothing but the image itself. Therefore, I would expect something useful, like the person's name.